Formula/embree.rb

```diff
@@ -17,7 +17,11 @@ class Embree < Formula
   depends_on "tbb"
 
   def install
-    max_isa = MacOS.version.requires_sse42? ? "SSE4.2" : "SSE2"
+    max_isa = "SSE4.2"
```

Comment:
On Linux `oldest_cpu` is `core2` which does not support SSE4.2, but does support SSE3.

> ‘core2’ Intel Core 2 CPU with 64-bit extensions, MMX, SSE, SSE2, SSE3 and SSSE3 instruction set support.
> ‘nehalem’ Intel Nehalem CPU with 64-bit extensions, MMX, SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2 and POPCNT instruction set support.

See https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
and https://en.wikipedia.org/wiki/List_of_Intel_CPU_microarchitectures

Reply:
I see this isn't bottled on Linux, and will probably have `cellar :any` when it is bottled, so it will probably never be built from source unless explicitly requested.  If our CI env supports `SSE4.2` but a user tries to install the bottle on a system which only support `SSE2`, would that result in a broken bottle?

Perhaps the solution here is to use `SSE2` on Linux so that the bottle works everywhere, and users who want better optimized instructions can build from source?  This is the first time I've seen this particular scenario so I'm not sure how we should handle it.

Reply:
> If our CI env supports SSE4.2 but a user tries to install the bottle on a system which only support SSE2, would that result in a broken bottle?

It depends on whether the executable detects at run time which instruction set is available, or if it always uses the instruction set that was specified at compile time. There's unfortunately no way to know without testing, or reading the upstream documentation.

Reply:
> Perhaps the solution here is to use `SSE2` on Linux so that the bottle works everywhere

Yes.

> and users who want better optimized instructions can build from source?

We don't support building from source so I'd rather we didn't point them in that direction.

Reply:
On a similar topic, with GitHub Packages it would be possible to upload bottles compiled for different Intel CPUs (different microarchitectures).

Reply:
We could've probably made micro-architecture specific bottles work with Bintray, no? The bigger impediment is figuring out how to make this work on CI.

Reply:
I've looked at the documentation for this now in more depth and it seems like this may use "smart" ISA checking where it lets you build binaries which can take advantage of newer ISAs but also checks at run time to make sure it only uses the supported ISAs for the CPU being used to execute.  OpenBLAS does something like this so that it doesn't have to be rebuilt for all the different microarchitectures it supports.

Right now the holdup from me testing it further is that the `ispc` dependency doesn't build in Linux CI so I have to fix that before I can investigate this further.

Reply:
If it does runtime ISA checks, then can we drop `max_isa` from this formula?

Reply:
That's what I'm thinking.  I need to test this on a server I have access to which has multiple generations of CPUs where I can try to build it with full set of supported ISAs (up to AVX-512) and then test on some older CPUs that definitely don't have that ISA.  I'm not sure if there is anything on macOS which limits ISA support beyond the hardware itself - I don't know what determines `MacOS.version.requires_sse42?`.

Reply:
`#requires_sse42?` is just a check for whether the oldest CPU that runs a given OS is older than Nehalem, I think: https://rubydoc.brew.sh/OS/Mac/Version.html#requires_nehalem_cpu%3F-instance_method

Though now that you mention OpenBLAS I'm wondering whether this is needed:

https://github.com/Homebrew/homebrew-core/blob/99f4b051013b379ebcb9d82d6acf35e6ab33068e/Formula/openblas.rb#L36

Reply:
That was probably related to some issues upstream had with the AVX-512 kernels not working correctly: https://github.com/xianyi/OpenBLAS/issues/2182.  However, those issues seem to have been fixed, so we should see if it works properly again.

Reply:
Looks like you have access to a good testing environment for that :)